src/Data/List/Relation/Binary/Permutation/Setoid/Properties.agda

```diff
@@ -29,44 +28,53 @@ open import Data.List.Relation.Unary.AllPairs using (AllPairs; []; _∷_)
 import Data.List.Relation.Unary.Unique.Setoid as Unique
 import Data.List.Membership.Setoid as Membership
 open import Data.List.Membership.Setoid.Properties using (∈-∃++; ∈-insert)
-import Data.List.Properties as Lₚ
+import Data.List.Properties as List
 open import Data.Nat hiding (_⊔_)
 open import Data.Nat.Induction
 open import Data.Nat.Properties
 open import Data.Product.Base using (_,_; _×_; ∃; ∃₂; proj₁; proj₂)
 open import Function.Base using (_∘_; _⟨_⟩_; flip)
 open import Level using (Level; _⊔_)
 open import Relation.Unary using (Pred; Decidable)
-import Relation.Binary.Reasoning.Setoid as RelSetoid
+import Relation.Binary.Reasoning.Setoid as ≈-Reasoning
 open import Relation.Binary.Properties.Setoid S using (≉-resp₂)
 open import Relation.Binary.PropositionalEquality.Core as ≡
   using (_≡_ ; refl; sym; cong; cong₂; subst; _≢_)
 open import Relation.Nullary.Decidable using (yes; no; does)
 open import Relation.Nullary.Negation using (contradiction)
 
-private
-  variable
-    b p r : Level
 
-open Setoid S using (_≈_) renaming (Carrier to A; refl to ≈-refl; sym to ≈-sym; trans to ≈-trans)
+open Setoid S using (_≈_)
+  renaming (Carrier to A; refl to ≈-refl; sym to ≈-sym; trans to ≈-trans)
 open Permutation S
 open Membership S
 open Unique S using (Unique)
 open module ≋ = Equality S
-  using (_≋_; []; _∷_; ≋-refl; ≋-sym; ≋-trans; All-resp-≋; Any-resp-≋; AllPairs-resp-≋)
-open PermutationReasoning
+  using (_≋_; []; _∷_
+        ; ≋-refl; ≋-sym; ≋-trans
+        ; All-resp-≋; Any-resp-≋; AllPairs-resp-≋)
+
+private
+  variable
+    b p r : Level
+    x y z v w : A
+    xs ys zs vs ws : List A
+    P : Pred A p
+    R : Rel A r
+
+  _++[_]++_ = λ xs (z : A) ys → xs ++ [ z ] ++ ys
```

Comment:
I have to confess I don't understand the purpose of this? As far as I can see it saves two space characters and that's it?

Reply:
Yeah, it's maybe taking DRY too far, but it's purely local. Elsewhere in #2317 though, I ended up having to use the qualified version of `_++_` and that was getting very heavy...

Easy to inline though!